Add common Cgroup interface

dqminh opened the pull request:
This is part of cgroupv2 patch set. Here we add a Cgroup interface that
both v1 and v2 need to conform to, and port cgroupv1 to use that first.

dqminh commented:
cc @avagin @fvoznika

avagin commented:
The patch looks good to me. CgroupJSON looks very familiar https://github.com/google/gvisor/commit/d85ae509c941ac055f6d8b1f46912b15cff530e6#diff-f552c5573a6d5130b7ba9d96a88bf82fb14a5749ae8f7b2baeffbd56d6dee33eR348.

avagin commented on runsc/cgroup/cgroup.go:
```diff
@@ -290,34 +291,47 @@ func loadPathsHelper(cgroup, mountinfo io.Reader) (map[string]string, error) {
 }
 
 // Cgroup represents a group inside all controllers. For example:
```
should we remove this line?

dqminh replied:
removed

dqminh commented:
> The patch looks good to me. CgroupJSON looks very familiar d85ae50#diff-f552c5573a6d5130b7ba9d96a88bf82fb14a5749ae8f7b2baeffbd56d6dee33eR348.

Yes ! i thought that i had changed the author to you, but i did not. This was changed now, and rebased on top of latest master.
